src/libs/actions/IOU.js

```diff
@@ -104,6 +106,14 @@ function buildOnyxDataForMoneyRequest(
         },
     ];
 
+    if (isNewChatReport && !_.isEmpty(optimisticPersonalDetailListAction)) {
+        optimisticData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.PERSONAL_DETAILS_LIST,
+            value: optimisticPersonalDetailListAction,
+        });
+    }
```

Comment:
This caused a [regression](https://github.com/Expensify/App/issues/30456). Merging the personal data should only be done only if it's not in Onyx already. Otherwise we would overwrite the existing user data e.g. displayName.